app/modules/encounters/resources.py

```diff
@@ -113,77 +113,80 @@ def patch(self, args, encounter):
         from app.extensions.elapsed_time import ElapsedTime
 
         timer = ElapsedTime()
-
         edm_args = [
             arg
             for arg in args
             if arg.get('path')
             in parameters.PatchEncounterDetailsParameters.PATH_CHOICES_EDM
         ]
-        if edm_args and len(edm_args) != len(args):
-            log.error(f'Mixed edm/houston patch called with args {args}')
-            abort(400, 'Cannot mix EDM patch paths and houston patch paths')
+        houston_args = [
+            arg
+            for arg in args
+            if arg.get('path')
+            in parameters.PatchEncounterDetailsParameters.PATH_CHOICES_HOUSTON
+        ]
+
+        if edm_args:
+            log.debug(f'wanting to do edm patch on args={edm_args}')
+            try:
+                result_data = current_app.edm.request_passthrough_result(
+                    'encounter.data',
+                    'patch',
+                    {'data': edm_args},
+                    encounter.guid,
+                    request_headers=request.headers,
+                )
+                # EDM uses id, houston API is all guid so ditch the id
+                result_data.pop('id', None)
+
+            except HoustonException as ex:
+                abort(
+                    ex.status_code,
+                    ex.message,
+                    error=ex.get_val('error', 'Error'),
+                    edm_status_code=ex.get_val('edm_status_code', None),
+                )
+        else:
+            # this mimics output format of edm-patching
+            result_data = {
+                'version': encounter.version,
+            }
 
-        if not edm_args:
+        if houston_args:
             context = api.commit_or_abort(
                 db.session, default_error_message='Failed to update Encounter details.'
             )
             with context:
                 try:
                     parameters.PatchEncounterDetailsParameters.perform_patch(
-                        args, obj=encounter
+                        houston_args, obj=encounter
                     )
                 except HoustonException as ex:
                     abort(
```

Comment:
am i reading correctly that this will do the `abort()` if the houston-side of the patch fails?  my understanding (which seems to be echoed in the ticket for this) is that _if_ there is an EDM patch which _has succeeded_ and _then_ the houston patch fails, we are not doing a full abort but rather reporting the "half success".  (though, granted, i am not exactly sure if we decided what this looks like!)

along these lines the above edm-failure-equals-total-failure logic _does_ seem correct (in line with ticket description).  i think the half-success is the weird special case now.